File: code/jankyLib/classes/jankyXboxJoystick.cpp
/******************************
 * JankyXboxJoystick.cpp      *
 *                            *
 *  Created on: Jan 21, 2013  *
 *  Author: Team1967          *
 ******************************/

#include "jankyXboxJoystick.h"
//#include "frc/WPILib.h"

/**********************************************************************************************
 * This is a demo program showing the use of the Joystick class. 
 * The Joystick class will automatically call the controls on the joystick in 
 * order to be applied to the robot. 
 * JankyXboxJoystick class uses specifically the Xbox controller.
 **********************************************************************************************/ 

/************************
* Default constructor  *
************************/
	
jankyXboxJoystick::jankyXboxJoystick(int portNumber) : Joystick (portNumber)
{
		
}

jankyXboxJoystick::~jankyXboxJoystick()
{
	
}

/**
 * Get button value (true or false) of button A (bottom button). 
 */
bool jankyXboxJoystick::GetButtonA()
{
	return Joystick::GetRawButton(BUTTON_A_PORT);
}

/**
 * Get button value (true or false) of button B (right most button). 
 */
bool jankyXboxJoystick::GetButtonB()
{
	return Joystick::GetRawButton(BUTTON_B_PORT);
}

/**
 * Get button value (true or false) of button X (left most button). 
 */
bool jankyXboxJoystick::GetButtonX()
{
	return Joystick::GetRawButton(BUTTON_X_PORT);
}

/**
 * Get button value (true or false) of button Y (top button). 
 */
bool jankyXboxJoystick::GetButtonY()
{
	if (Joystick::GetRawButton(BUTTON_Y_PORT))
	{	
		return true;
	}
	else 
		return false;
	
	//return Joystick::GetRawButton(BUTTON_Y_PORT);	
}

/**
 * Get button value (true or false) of button LB. 
 */
bool jankyXboxJoystick::GetButtonLB()
{
	return Joystick::GetRawButton(BUTTON_LB_PORT);
}

/**
 * Get button value (true or false) of button RB. 
 */
bool jankyXboxJoystick::GetButtonRB()
{
	return Joystick::GetRawButton(BUTTON_RB_PORT);
}

/**
 * Get button value (true or false) of button BACK. 
 */
bool jankyXboxJoystick::GetButtonBack()
{
	return Joystick::GetRawButton(BUTTON_BACK_PORT);
}

/**
 * Get button value (true or false) of button START. 
 */
bool jankyXboxJoystick::GetButtonStart()
{
	return Joystick::GetRawButton(BUTTON_START_PORT);
}

/**
 * Get button value (true or false) of the left thumbstick button.
 */
bool jankyXboxJoystick::GetLeftStickButton()
{
	return Joystick::GetRawButton(LEFT_STICK_BUTTON_PORT);
}

/**
 * Get button value (true or false) of the right thumbstick button.
 */
bool jankyXboxJoystick::GetRightStickButton()
{
	return Joystick::GetRawButton(RIGHT_STICK_BUTTON_PORT);
}

/**
 * Returns float from left trigger (0.0-1.0)
 */
float jankyXboxJoystick::GetLeftThrottle() 
{
	float leftThrottle = GetTwist();
	if (leftThrottle > 0.2) {
		return leftThrottle;
	}
	else return 0;
}

/**
 * Returns float from right trigger (0.0-1.0)
 */
float jankyXboxJoystick::GetRightThrottle() 
{
	float rightThrottle = GetThrottle();
	if (rightThrottle > 0) {
		return rightThrottle;
	}
	else return 0;
}

/**
 * Get the state of the left thumbstick axis. Returns a float value.
 */
float jankyXboxJoystick::GetLeftXAxis ()
{
	float x = GetRawAxis(LEFT_X_AXIS_CHANNEL);
	return x;
}

/**
 * Get the state of the right thumbstick axis. Returns a float value.
 */
float jankyXboxJoystick::GetLeftYAxis()
{
	float y = (GetRawAxis(LEFT_Y_AXIS_CHANNEL));
	return y;
}

float jankyXboxJoystick::GetRightXAxis ()
{
	float x = GetRawAxis(RIGHT_X_AXIS_CHANNEL);
	return x;
}

float jankyXboxJoystick::GetRightYAxis ()
{
	float y = (GetRawAxis(RIGHT_Y_AXIS_CHANNEL));
	return y;
}
float jankyXboxJoystick::GetLeftTwist()
{
	float leftTwist = GetTwist();
	return leftTwist;	
}
float jankyXboxJoystick::GetRightTwist()
{
	float rightTwist = GetTwist();
	return rightTwist;	
}

File: code-java/jankyLib/jankyXboxJoystick.java

package org.janksters.jankyLib;
import edu.wpi.first.wpilibj.Joystick;
//import edu.wpi.first.wpilibj.GenericHID;
 
 
public class jankyXboxJoystick extends Joystick {
   final static int BUTTON_A_PORT =1;
   final static int BUTTON_B_PORT =2;
   final static int BUTTON_X_PORT =3;
   final static int BUTTON_Y_PORT =4;
   final static int BUTTON_LB_PORT =5;
   final static int BUTTON_RB_PORT =6;
   final static int BUTTON_BACK_PORT =7;
   final static int BUTTON_START_PORT =8;
   final static int LEFT_STICK_BUTTON_PORT =9;
   final static int RIGHT_STICK_BUTTON_PORT =10;
   final static int LEFT_X_AXIS_CHANNEL =0;
   final static int LEFT_Y_AXIS_CHANNEL =1;
   final static int RIGHT_X_AXIS_CHANNEL=4;
   final static int RIGHT_Y_AXIS_CHANNEL =5;
 
   //public jankyXboxJoystick(int portNumber);
   public jankyXboxJoystick (int portNumber){
       super(portNumber);
 
   }
 
   //jankyXboxJoystick XboxController = new jankyXboxJoystick(2);
  
 
   public boolean GetButtonA(){
       return getRawButton(BUTTON_A_PORT);
   }
  
   public boolean GetButtonB(){
       return getRawButton(BUTTON_B_PORT);
   }
 
   public boolean GetButtonX(){
       return getRawButton(BUTTON_X_PORT);
   }
   public boolean GetButtonY(){
       if (getRawButton(BUTTON_Y_PORT)) return true;
       else return false;
   }
   public boolean GetButtonLB(){
       return getRawButton(BUTTON_LB_PORT);
   }
   public boolean GetButtonRB(){
       return getRawButton(BUTTON_RB_PORT);
   }
   public boolean GetButtonBack(){
       return getRawButton(BUTTON_BACK_PORT);
   }
   public boolean GetButtonStart(){
       return getRawButton(BUTTON_START_PORT);
   }
   public boolean GetLeftStickButton(){
       return getRawButton(LEFT_STICK_BUTTON_PORT);
   }
   public boolean GetRightStickButton(){
       return getRawButton(RIGHT_STICK_BUTTON_PORT);
   }
 
   public double GetLeftThrottle(){
       double leftThrottle = getTwist();
       if (leftThrottle > 0.2){
           return leftThrottle;
       }
       else return 0;
   }
   public double GetRightThrottle(){
       double rightThrottle = getThrottle();
       if (rightThrottle>0){
           return rightThrottle;
       }
       else return 0;
   }
   //C++ has an argument but java seems to hate that
   //public float GetLeftXAxis(){
   //    float x = GetLeftXAxis();
   //    return x;
   //}
   public double GetLeftYAxis(){
       double y = getRawAxis(LEFT_Y_AXIS_CHANNEL);
       return y;
   }
   public double GetRightXAxis(){
       double x= getRawAxis(RIGHT_X_AXIS_CHANNEL);
       return x;
   }
 
   public double GetRightYAxis(){
       double y=(getRawAxis(RIGHT_Y_AXIS_CHANNEL));
       return y;
   }
   public double GetLeftTwist(){
       double leftTwist = getTwist();
       return leftTwist;
   }
   public double GetRightTwist(){
       double rightTwist = getTwist();
       return rightTwist;
   }
}